executor.go

```diff
@@ -0,0 +1,139 @@
+package goptions
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+)
+
+var (
+	//ErrNoCommand This should never happen.
```

Comment:
This comment should be more in the shape of `ErrNoCommand should never be returned`. 

But more importantly: If it is never supposed to be returned, why return it? Seems like an appropriate place for a `panic()` imo.